flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MultipleProgramsTestBase.java

```diff
@@ -80,29 +80,36 @@
 	
 	protected final TestExecutionMode mode;
 
-	
+	private TestEnvironment testEnvironment;
+
+	private CollectionTestEnvironment collectionTestEnvironment;
+
 	public MultipleProgramsTestBase(TestExecutionMode mode) {
 		this.mode = mode;
-		
+
 		switch(mode){
 			case CLUSTER:
-				new TestEnvironment(cluster, 4).setAsContext();
+				testEnvironment = new TestEnvironment(cluster, 4);
```

Comment:
Could you try this instead: https://github.com/junit-team/junit4/issues/45#issuecomment-6422143